src/odemis/driver/xt_client.py

```diff
@@ -870,125 +909,45 @@ def __init__(self, name, role, parent, hfw_nomag, **kwargs):
         self.depthOfField = model.FloatContinuous(1e-6, range=(0, 1e3),
                                                   unit="m", readonly=True)
         self._updateDepthOfField()
+        rng_width = self.parent.resolution_info()["range"]["width"]
+        rng_height = self.parent.resolution_info()["range"]["height"]
+        self._shape = (rng_width[1], rng_height[1])
+        # pixelSize is the same as MD_PIXEL_SIZE, with scale == 1
+        # == smallest size/ between two different ebeam positions
+        pxs = (self._hfw_nomag / (self._shape[0] * mag),
+               self._hfw_nomag / (self._shape[0] * mag))
+        self.pixelSize = model.VigilantAttribute(pxs, unit="m", readonly=True)
+
+        # .resolution is the number of pixels actually scanned. If it's less than
+        # the whole possible area, it's centered.
+        resolution = self.parent.get_resolution()
+
+        self.resolution = model.ResolutionVA(tuple(resolution),
+                                             ((rng_width[0], rng_height[0]),
+                                              (rng_width[1], rng_height[1])),
+                                             setter=self._setResolution)
+        self._resolution = resolution
+
+        # (float, float) as a ratio => how big is a pixel, compared to pixelSize
+        # it basically works the same as binning, but can be float
+        # (Default to scan the whole area)
+        self._scale = (self._shape[0] / resolution[0], self._shape[1] / resolution[1])
+        self.scale = model.TupleContinuous(self._scale,
+                                           [(1, 1), self._shape],
+                                           unit="",
+                                           cls=(int, float),  # int; when setting scale the GUI returns a tuple of ints.
+                                           setter=self._setScale)
+        self.scale.subscribe(self._onScale, init=True)  # to update metadata
+
+        # If scaled up, the pixels are bigger
+        pxs_scaled = (pxs[0] * self.scale.value[0], pxs[1] * self.scale.value[1])
+        self.parent._metadata[model.MD_PIXEL_SIZE] = pxs_scaled
 
```

Comment:
BTW, I've noticed that the API provides a "get_DeltaFWD()" function, which should return the value for a "depthOfField" VA. It's not necessary, but it'd good to keep in mind, especially that is helpful for manual focus. It'd probably would have to be updated every time the horizontalFoV is changed.